src/link/main.c

```diff
@@ -283,9 +283,9 @@ static void parseScrambleSpec(char const *spec)
 
 		while (region < SCRAMBLE_UNK) {
 			// If the strings match (case-insensitively), we got it!
-			// It's OK not to use `strncasecmp` because `regionName` is still
-			// NUL-terminated, since the encompassing spec is.
-			if (!strcasecmp(scrambleSpecs[region].name, regionName))
+			// `strncasecmp` must be used here since `regionName` is not
+			// NUL-terminated and points to the entire remaining argument.
+			if (!strncasecmp(scrambleSpecs[region].name, regionName, regionNameLen))
 				break;
 			region++;
 		}
```

Comment:
Using a `for` loop for brevity, and don't say `regionName` is not NUL-terminated (it is, just not where we want).
```c
		for (; region < SCRAMBLE_UNK; region++) {
			// `strncasecmp` must be used here since `regionName` points
			// to the entire remaining argument.
			if (!strncasecmp(scrambleSpecs[region].name, regionName, regionNameLen))
				break;
		}
```